tests/digraph/test_bellman_ford.py

```diff
@@ -0,0 +1,316 @@
+            (self.c, self.f, 11),
+            (self.e, self.f, 6),
+        ]
+        self.graph.add_edges_from(edge_list)
+
+    def test_bellman_ford(self):
+        path = retworkx.digraph_bellman_ford_shortest_path_lengths(
+            self.graph, self.a, lambda x: float(x)
+        )
+        expected = {1: 7.0, 2: 16.0, 3: 14.0, 4: 23.0, 5: 22.0}
+        self.assertEqual(expected, path)
+
+    def test_bellman_ford_length_with_no_path(self):
+        g = retworkx.PyDiGraph()
+        a = g.add_node("A")
+        g.add_node("B")
+        path_lenghts = retworkx.digraph_bellman_ford_shortest_path_lengths(g, a, edge_cost_fn=float)
+        expected = {}
+        self.assertEqual(expected, path_lenghts)
+
+    def test_bellman_ford_path(self):
+        paths = retworkx.digraph_bellman_ford_shortest_paths(self.graph, self.a)
+        expected = {
+            # a -> b
+            1: [0, 1],
+            # a -> c: a, d, c
+            2: [0, 3, 2],
+            # a -> d
+            3: [0, 3],
+            # a -> e: a, d, e
+            4: [0, 3, 4],
+            # a -> f: a, b, f
+            5: [0, 1, 5],
+        }
+        self.assertEqual(expected, paths)
+
+    def test_bellman_ford_path_with_weight_fn(self):
+        paths = retworkx.digraph_bellman_ford_shortest_paths(
+            self.graph, self.a, weight_fn=lambda x: x
+        )
+        expected = {
+            1: [0, 1],
+            2: [0, 3, 2],
+            3: [0, 3],
+            4: [0, 3, 4],
+            5: [0, 1, 5],
+        }
+        self.assertEqual(expected, paths)
+
+    def test_bellman_ford_path_undirected(self):
+        paths = retworkx.digraph_bellman_ford_shortest_paths(self.graph, self.a, as_undirected=True)
+        expected = {
+            1: [0, 1],
+            2: [0, 2],
+            3: [0, 3],
+            4: [0, 3, 4],
+            5: [0, 1, 5],
+        }
+        self.assertEqual(expected, paths)
+
+    def test_bellman_ford_path_undirected_with_weight_fn(self):
+        paths = retworkx.digraph_bellman_ford_shortest_paths(
+            self.graph, self.a, weight_fn=lambda x: x, as_undirected=True
+        )
+        expected = {
+            1: [0, 1],
+            2: [0, 2],
+            3: [0, 2, 3],
+            4: [0, 2, 3, 4],
+            5: [0, 2, 5],
+        }
+        self.assertEqual(expected, paths)
+
+    def test_bellman_ford_with_no_goal_set(self):
+        path = retworkx.digraph_bellman_ford_shortest_path_lengths(self.graph, self.a, lambda x: 1)
+        expected = {1: 1.0, 2: 2.0, 3: 1.0, 4: 2.0, 5: 2.0}
+        self.assertEqual(expected, path)
+
+    def test_bellman_path(self):
+        path = retworkx.digraph_bellman_ford_shortest_paths(
+            self.graph, self.a, weight_fn=lambda x: float(x), target=self.e
+        )
+        expected = retworkx.digraph_dijkstra_shortest_paths(
+            self.graph, self.a, weight_fn=lambda x: float(x), target=self.e
+        )
+        self.assertEqual(expected, path)
```

Comment:
What's the purpose of this test?

Reply:
It tests the target argument. Because I copied it from the Dijkstra's test, I compared it to Dijkstra's result instead of a hardcoded result. I think it it is valid because for any graph with non-negative weights, Dijkstra/Bellman-Ford/Floyd-Warshall should give the same answer